### ISSUE TITLE
[CLOSED] COE (from bundle 0.0.4) attempts to use functions fmi2GetMaxStepsize() - might cause crash

### ISSUE THREAD

**Issue by ldcouto** [into-cps/intocps-ui#89]
<a href="https://github.com/omoeller"><img src="https://avatars.githubusercontent.com/u/11794804?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [omoeller](https://github.com/omoeller)**
_Tuesday Feb 09, 2016 at 10:59 GMT_
_Originally opened as https://github.com/into-cps/INTO-CPS_Application/issues/4_

----

Observed with (md5sum)
ddb44a7017102573b48691f974cfa5eb coe-0.0.1-SNAPSHOT-jar-with-dependencies.jar
(from bundle 0.0.4)
on a Windows8 System (64bit) running a 64bit Java VM:

When loading the tank.fmu / tankcontroller.fmu (from bundle 0.0.4)
apparently the COE searches for function fmi2GetMaxStepsize

...
 INFO [Thread-2] (CoeInitialize.scala:220) -    {8c4e810f-3df3-4a00-8276-176fa3c9f000}.fmu1.valve --> valve
warning: Function fmi2GetMaxStepsize not found in dll

I had a look at the DLL: true, no such function in there.

However, this function is not mentioned in the FMI2.0 standard, maybe its an 
experimental addition. 

At least for me, the COE crashes in this experiment.
My guess is that the COE tries to call fmi2GetMaxStepsize() anyway.

## -- Here is the full client output. -----------------

c:/Users/oli/RTT-Projects/utils/run-COE.py tank.fmu tankcontoller.fmu

Sending command "/initialize" with payload
{
    "connections": {
        "{8c4e810f-3df3-4a00-8276-176fa3c9f001}.fmu0.level": [
            "{8c4e810f-3df3-4a00-8276-176fa3c9f000}.fmu1.level"
        ],
        "{8c4e810f-3df3-4a00-8276-176fa3c9f000}.fmu1.valve": [
            "{8c4e810f-3df3-4a00-8276-176fa3c9f001}.fmu0.valve"
        ]
    },
    "algorithm": {
        "type": "fixed-step",
        "size": 0.1
    },
    "parameters": {},
    "fmus": [
        "E:\\tank.fmu",
        "E:\\tankcontroller.fmu"
    ]
}
Received response:
[
    {
        "status": "Initialized",
        "avaliableLogLevels": {
            "{8c4e810f-3df3-4a00-8276-176fa3c9f001}.fmu0": [
                {
                    "name": "logAll",
                    "description": null
                },
                {
                    "name": "logError",
                    "description": null
                },
                {
                    "name": "logFmiCall",
                    "description": null
                },
                {
                    "name": "logEvent",
                    "description": null
                }
            ],
            "{8c4e810f-3df3-4a00-8276-176fa3c9f000}.fmu1": [
                {
                    "name": "logAll",
                    "description": null
                },
                {
                    "name": "logError",
                    "description": null
                },
                {
                    "name": "logFmiCall",
                    "description": null
                },
                {
                    "name": "logEvent",
                    "description": null
                }
            ]
        },
        "sessionId": 73885,
        "lastExecTime": 0
    }
]
application/json
Sending command "/simulate/73885" with payload
{
    "logLevels": {
        "{8c4e810f-3df3-4a00-8276-176fa3c9f001}.fmu0": [
            "logAll",
            "logError",
            "logFmiCall",
            "logEvent"
        ],
        "{8c4e810f-3df3-4a00-8276-176fa3c9f000}.fmu1": [
            "logAll",
            "logError",
            "logFmiCall",
            "logEvent"
        ]
    },
    "endTime": 10.0,
    "startTime": 0.0
}
Traceback (most recent call last):
  File "c:/Users/oli/RTT-Projects/utils/run-COE.py", line 259, in <module>
    coe.simulate()
  File "c:/Users/oli/RTT-Projects/utils/run-COE.py", line 202, in simulate
    return self.sendCommand(url, payload = payload)
  File "c:/Users/oli/RTT-Projects/utils/run-COE.py", line 167, in sendCommand
    res = conn.getresponse()
  File "C:\Python27\lib\httplib.py", line 1132, in getresponse
    response.begin()
  File "C:\Python27\lib\httplib.py", line 453, in begin
    version, status, reason = self._read_status()
  File "C:\Python27\lib\httplib.py", line 409, in _read_status
    line = self.fp.readline(_MAXLINE + 1)
  File "C:\Python27\lib\socket.py", line 480, in readline
    data = self._sock.recv(self._rbufsize)
socket.error: [Errno 10054] An existing connection was forcibly closed by the remote host

## -- Here is the full COE output. --------------------

```
oli@drake-vm /cygdrive/c/opt/coe
$ java -jar coe-0.0.1-004.SNAPSHOT-jar-with-dependencies.jar
COE(Command line options: [port]

 INFO [main] (CoeMain.java:29) - Checking native FMI support
Now running on port 8082
Hit Enter to stop.

 INFO [Thread-1] (CoeRequestHandler.java:257) - Using Fixed-step size calculator
 with size = 0.1
[E:\tank.fmu, E:\tankcontroller.fmu]
[{8c4e810f-3df3-4a00-8276-176fa3c9f001}.fmu0.level -> {8c4e810f-3df3-4a00-8276-1
76fa3c9f000}.fmu1.level, {8c4e810f-3df3-4a00-8276-176fa3c9f000}.fmu1.valve -> {8
c4e810f-3df3-4a00-8276-176fa3c9f001}.fmu0.valve]
[]
 INFO [Thread-1] (CoeInitialize.scala:54) - loading fmus
TRACE [Thread-1] (CoeInitialize.scala:209) - Set({8c4e810f-3df3-4a00-8276-176fa3c9f001}.fmu0, {8c4e810f-3df3-4a00-8276-176fa3c9f000}.fmu1)
 INFO [Thread-1] (CoeInitialize.scala:213) - Outputs from: {8c4e810f-3df3-4a00-8276-176fa3c9f001}.fmu0 = Set(level)
 INFO [Thread-1] (CoeInitialize.scala:213) - Outputs from: {8c4e810f-3df3-4a00-8276-176fa3c9f000}.fmu1 = Set(valve)
 INFO [Thread-1] (CoeInitialize.scala:218) - Inputs for: {8c4e810f-3df3-4a00-8276-176fa3c9f000}.fmu1
 INFO [Thread-1] (CoeInitialize.scala:220) -    {8c4e810f-3df3-4a00-8276-176fa3c9f001}.fmu0.level --> level
 INFO [Thread-1] (CoeInitialize.scala:218) - Inputs for: {8c4e810f-3df3-4a00-8276-176fa3c9f001}.fmu0
 INFO [Thread-1] (CoeInitialize.scala:220) -    {8c4e810f-3df3-4a00-8276-176fa3c9f000}.fmu1.valve --> valve
warning: Function fmi2GetMaxStepsize not found in dll
warning: Function fmi2GetMaxStepsize not found in dll
TRACE [Thread-1] (BasicFixedStepSizeCalculator.scala:15) - Initializing the fixed step size calculator
TRACE [Thread-1] (CoeInitialize.scala:89) - Obtaining logging categories from the modelDescirption.xml files
TRACE [Thread-1] (CoeInitialize.scala:94) - Initialization done
TRACE [Thread-1] (CoeRequestHandler.java:326) - Initialization completed obtained the following logging categories: {{8c4e810f-3df3-4a00-8276-176fa3c9f001}.fmu0=[logAll, logError, logFmiCall, logEvent], {8c4e810f-3df3-4a00-8276-176fa3c9f000
}.fmu1=[logAll, logError, logFmiCall, logEvent]}
logAll
logError
logFmiCall
logEvent
returned from fmu
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  EXCEPTION_ACCESS_VIOLATION (0xc0000005) at pc=0x00000000777c32f2, pid=4472, tid=3088
#
# JRE version: Java(TM) SE Runtime Environment (8.0_74-b02) (build 1.8.0_74-b02)

# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.74-b02 mixed mode windows-amd64 compressed oops)
# Problematic frame:
# C  [ntdll.dll+0x532f2]
#
# Failed to write core dump. Minidumps are not enabled by default on client versions of Windows
#
# An error report file with more information is saved as:
# C:\opt\coe\hs_err_pid4472.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
```

---
oli@drake-vm /cygdrive/c/opt/coe
$ java -version
java version "1.8.0_74"
Java(TM) SE Runtime Environment (build 1.8.0_74-b02)
Java HotSpot(TM) 64-Bit Server VM (build 25.74-b02, mixed mode)

**Comment by ldcouto** [comment]
<a href="https://github.com/CThuleHansen"><img src="https://avatars.githubusercontent.com/u/3479481?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [CThuleHansen](https://github.com/CThuleHansen)**
_Tuesday Mar 01, 2016 at 14:53 GMT_

----

hi oliver, we will be releasing a new version shortly. 
If possible, we would like you to try this new version.
We do not invoke the function fmi2GetMaxStepsize if it is not present in the dll.

**Comment by ldcouto** [comment]
<a href="https://github.com/omoeller"><img src="https://avatars.githubusercontent.com/u/11794804?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [omoeller](https://github.com/omoeller)**
_Tuesday Mar 01, 2016 at 21:37 GMT_

----

hi Casper. 
it was just a suspicion that the crash might be due to fmi2GetMaxStepsize(); maybe one of the FMUs released with baseline 0.0.4 just has faulty code and causes a segfault.
